### ISSUE TITLE
Will LibLog be replaced?

### ISSUE THREAD

**Issue by HolisticDeveloper** [HangfireIO/Hangfire#1584]
I just read that LibLog has been deprecated (see https://github.com/damianh/LibLog/issues/270). Has any thought be given yet to whether Hangfire will switch to Microsoft.Extensions.Logging or a different logging abstraction?

**Comment by odinserj** [comment]
The rule #1 for logging in a framework – don't depend on any external package. Otherwise sooner or later there will be a breaking change in that package, and you'll have a version lock, where you can't upgrade without breaking all the current installations, and can't leave the current version, because newer projects will be unable to use it.

LibLog plays well as it's a simple source file and doesn't force you to have any external NuGet package references.

**Comment by kilroyFR** [comment]
Agreed with odinserj, microsoft.extension.logging is wired with others nugets, requires modification on all constructors for same functionality. Totally useless. On some of my personal projects i sticked on liblog.

**Comment by stijnherreman** [comment]
@odinserj any chance of reconsidering the use of Microsoft.Extensions.Logging? If not, I'll try upgrading the LibLog dependency to the latest version.

**Comment by odinserj** [comment]
Unfortunately, there are breaking changes in the LibLog of the latest version, so it's currently not possible just to upgrade it. `Hangfire.NetCore` package provides an adapter for `Microsoft.Extensions.Logging`, so it works with it. One problem is current logging is the absence of structured logging, but anyway it should be addressed without introducing breaking changes.

**Comment by odinserj** [comment]
Also, LibLog is added to Hangfire as sources, and there's no dependency on its NuGet package.
